### PR TITLE
test: Added CI for pyodide

### PR DESCRIPTION
**Pull request by andeplane**
## Description
Closes https://github.com/cognitedata/cog-ai/issues/2336

The Python SDK is heavily used in our Jupyter notebook and Streamlit environment which is Pyodide based. To ensure compatibility, I have added a CI step that verifies that the SDK can be installed and imported in Pyodide.

The test is implemented as a nodejs script triggered by GitHub workflows, one for Jupyter and one for Streamlit.

## Checklist:
- [x] Tests added/updated.
- [ ] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [ ] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [ ] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

**Comment by haakonvt** [comment]
What are the differences now between test jupyter and test streamlit?

**Comment by andeplane** [comment]
> What are the differences now between test jupyter and test streamlit?

Now it's only a minor version difference on Pyodide. But we will add the different set of packages they use in addition.

File: .github/workflows/verify-jupyter.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+on:
+  pull_request:
+    branches: [master]
+
+jobs:
+  build_and_test_jupyter_pyodide:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/setup
```
[the remark this answers is not in the document]
Comment (andeplane) [reply .github/workflows/verify-jupyter.yml:13]: Don't we need that for building the SDK?

[the remark this answers is not in the document]
Comment (andeplane) [reply .github/workflows/verify-jupyter.yml:13]: I also see that I don't run `poetry install`, not sure why this works :D

[the remark this answers is not in the document]
Comment (haakonvt) [reply .github/workflows/verify-jupyter.yml:13]: > Don't we need that for building the SDK?
Oh, right, yeah for sure needed with poetry at least

**Review by haakonvt — APPROVED** [review]
LGTM 🚀

**Pull request merged**